File: presentation/src/main/java/com/crowleysimon/current/extensions/SaturateExtension.java
package com.crowleysimon.current.extensions;

import android.graphics.drawable.Drawable;
import androidx.annotation.NonNull;
import com.bumptech.glide.RequestBuilder;
import com.bumptech.glide.annotation.GlideExtension;
import com.bumptech.glide.annotation.GlideType;
import com.bumptech.glide.load.resource.drawable.DrawableTransitionOptions;

@GlideExtension
public class SaturateExtension {

    private SaturateExtension() { }

    @NonNull
    @GlideType(Drawable.class)
    public static RequestBuilder<Drawable> saturateOnLoad(RequestBuilder<Drawable> requestBuilder) {
        return requestBuilder.transition(DrawableTransitionOptions.with(new SaturationTransitionFactory()));
    }
}

File: presentation/src/main/java/com/crowleysimon/current/extensions/SaturateExtension.kt
package com.crowleysimon.current.extensions

import android.graphics.drawable.Drawable
import com.bumptech.glide.RequestBuilder
import com.bumptech.glide.annotation.GlideExtension
import com.bumptech.glide.annotation.GlideType
import com.bumptech.glide.load.resource.drawable.DrawableTransitionOptions

@GlideExtension
object SaturateExtension {

    @GlideType(Drawable::class)
    @JvmStatic
    fun saturateOnLoad(requestBuilder: RequestBuilder<Drawable>): RequestBuilder<Drawable> {
        return requestBuilder.transition(DrawableTransitionOptions.with(SaturationTransitionFactory()))
    }
}
